lib/block-list.js

```diff
@@ -32,5 +34,25 @@ function renderItem (item) {
 }
 
 export default function ({body}) {
-  return body.map(renderItem).filter(Boolean);
+  const bundlesToUrls = {};
+  const urlsToBundles = {};
+  let count = 0;
+
+  const components = body
+    .map(renderItem)
+    .filter(Boolean);
+
+  components.forEach(item => {
+    if (item.components && item.components[0] && item.components[0].role === 'photo') {
+      const url = item.components[0].URL;
+      if (!urlsToBundles[url]) {
+        urlsToBundles[url] = `image-${count}`;
+        bundlesToUrls[`image-${count}`] = url;
```

Comment:
Hm... Ok. So we sometimes have png images that have `.jpg` at the end. Or at least we used to have that problem... What's the best solution here? Go through the urls and identify their content type before uploading?